File: shazam_export.php
<?php

require __DIR__ . '/vendor/autoload.php';

use GuzzleHttp\Client;

$client = new Client();
$dotenv = new Dotenv\Dotenv(__DIR__);

// Load from .env file
$dotenv->load();
$endpoint = 'https://amp.shazam.com/shazam/v1/en-US/BR/android/-/installation/' . getenv('SHAZAM_ID') . '/tagevents';
$apiKey = getenv('SHAZAM_KEY');

$allTags = [];

// Get tags...
echo 'Getting tags...';

do {
    $tags = getTags($endpoint, $apiKey, $client);

    foreach ($tags['events'] as $tag) {
        array_push($allTags, $tag);
    }

    echo '.';

    $endpoint = isset($tags['next']['url']) ? $tags['next']['url'] : null;
} while ($endpoint != null);

echo "\n";

// Encode and save to file...
saveJsonToFile('tags', $allTags);

echo count($allTags) . " tags found!\n";

// Get info for each track...
for ($i = 0; $i < count($allTags); $i++) {
    $track = getTrack($allTags[$i]['tag']['key'], $client);
    $allTags[$i]['track'] = $track;
    echo 'Getting tracks info: ' . $i . '/' . count($allTags) . "\n";
}

saveJsonToFile('tags_tracks', $allTags);

// Get tag list from the user...
function getTags($endpoint, $apiKey, $client) {
    $res = $client->request('GET', $endpoint, [
        'headers' => [
            'x-shazam-ampkey' => $apiKey,
        ]
    ]);
    return json_decode($res->getBody(), true);
}

// Get track metadata.
function getTrack($trackId, $client) {
    $res =  $client->request('GET', 'https://cdn.shazam.com/discovery/v4/en-US/BR/android/-/track/' . $trackId);
    return json_decode($res->getBody(), true);
}

// Save tags to json file.
function saveJsonToFile($fileName, $tags) {
    $serializedTags = json_encode($tags);
    $fileName = $fileName . '_' . time() . '.json';
    $file = fopen($fileName, 'w');
    fwrite($file, $serializedTags);
    fclose($file);
}

File: shazam_export.rb
require 'rest-client'
require 'json'

url = 'https://amp.shazam.com/shazam/v1/en-US/BR/android/-/installation/__USER__/tagevents'
key = ''
track_url = 'https://cdn.shazam.com/discovery/v4/en-US/BR/android/-/track/'

shazam_tags = []

puts 'Getting TAG data...'

loop do
    result = RestClient.get(url, {
        'x-shazam-ampkey': key
    })

    parsed_result = JSON.parse(result)

    tags = parsed_result['events']
    url = parsed_result['next']['url']

    tags.each do |tag|
        shazam_tags.push(tag)
        puts tag['tagid']
    end

    break if tags.length == 0
end

# puts shazam_tags

puts 'Getting tracks data...'

for i in 0..shazam_tags.length - 1 do
    result = RestClient.get(track_url + shazam_tags[i]['tag']['key'])
    parsed_result = JSON.parse(result)

    shazam_tags[i]['track'] = parsed_result

    puts i.to_s + '/' + shazam_tags.length.to_s + ' tracks added'
end

# puts shazam_tags.to_json

puts 'Saving to file...'

filename = 'tags_tracks_' + Time.now.getutc.to_i.to_s + '.json'
File.open(filename, 'w') { |file| file.write(shazam_tags.to_json) }

puts 'Done!'
